src/index.ml

```diff
@@ -332,14 +368,38 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
       Log.debug (fun l ->
           l "[%s] no changes detected" (Filename.basename t.root))
     in
-    (match t.log with None -> try_load_log t | Some _ -> ());
+    let add_log_entry log e = Tbl.replace log.mem e.key e.value in
+    ( match t.log with
+    | None -> t.log <- try_load_log t (log_path t.root)
+    | Some _ -> () );
+    ( match t.log_async with
+    | None -> t.log_async <- try_load_log t (log_async_path t.root)
+    | Some log -> (
+        try
+          let log_offset = IO.offset log.io in
+          IO.close log.io;
+          let path = log_async_path t.root in
+          if Sys.file_exists path then (
+            let io =
+              IO.v ~fresh:false ~readonly:true ~generation:0L ~fan_size:0L path
+            in
+            t.log_async <- Some { log with io };
+            let new_log_offset = IO.offset io in
+            if log_offset <> new_log_offset then (
+              Tbl.clear log.mem;
+              iter_io (add_log_entry log) io ) )
+          else ()
+        with IO.Bad_Fd_Read ->
```

Comment:
Perhaps this exception should not have `Fd` in its name: that's a detail of the  `Index_unix` usecase. It also hasn't been documented / explained anywhere (e.g. in `src/io.mli`).